### DISCUSSION TITLE
Logic error in dark theme behaviour example

### DISCUSSION THREAD

**Discussion by lukezak** [mantinedev/mantine#550] [Feedback]
When implementing the example 'Save to localStorage and add keyboard shortcut' under 'Dark theme' in the docs, the following statement throws the error `this condition will always return 'false' since the types '"light"' and '"dark"' have no overlap.`

`setColorScheme(value || (colorScheme === 'dark' ? 'light' : 'dark'));`

**Comment by rtivital** [comment]
You need to specify type in use-local-storage hook:
```tsx
const [colorScheme, setColorScheme] = useLocalStorageValue<'light | 'dark>({
    key: 'mantine-color-scheme',
    defaultValue: 'light',
  });
```
